File: backend/app/api/market.py
from fastapi import APIRouter

from app.services.market_env import compute_regime, get_breadth, get_indices, get_sectors

router = APIRouter()


@router.get("/regime")
async def market_regime():
    return compute_regime()


@router.get("/indices")
async def market_indices():
    return {"indices": get_indices()}


@router.get("/breadth")
async def market_breadth():
    return get_breadth()


@router.get("/sectors")
async def market_sectors():
    return {"sectors": get_sectors()}

File: frontend/src/api/market.js
import { createApiClient } from './http.js'

const api = createApiClient('/market')

export default {
  regime() {
    return api.get('/regime').then((r) => r.data)
  },
  indices() {
    return api.get('/indices').then((r) => r.data)
  },
  breadth() {
    return api.get('/breadth').then((r) => r.data)
  },
  sectors() {
    return api.get('/sectors').then((r) => r.data)
  }
}
